### ISSUE TITLE
`torch.compile` should result in an optimized module where `module.training` is the same as in the unoptimized module

### ISSUE THREAD

**Issue by uwu-420** [pytorch/pytorch#122414]
### 🚀 The feature, motivation and pitch

Hi, basically what the title says.
The current behavior of `torch.compile` is imo quite unexpected and can lead users to the false belief that a model is in eval mode.

### Alternatives

Alternatively, it would be a good idea to add to the documentation of `torch.compile` that the resulting optimized module always is in train mode.

cc @ezyang @msaroufim @bdhirsh @anijain2305 @zou3519 @chauhang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng

**Comment by jansel** [comment]
cc @anijain2305 I think you wrote that wrapper.

We could pretty easily change the value returned, though we might need to error if the user tries to change it (or forward that change to the original module).
